lib/linked_in/api.rb

```diff
@@ -124,5 +128,34 @@ def no_access_token_error
       msg = LinkedIn::ErrorMessages.no_access_token
       LinkedIn::InvalidRequest.new(msg)
     end
+
+    def verify_api_version!(version)
+      @api_version = if version.nil?
+        recent_api_version
+      else
+        validate_api_version!(version)
+      end
+    end
+
+    ## Verify the LinkedIn API version provided from caller.
+    # LinkedIn Marketing API Program will publish new versions monthly,
+    # and those versions will be supported for a minimum of one year.
+    # Raise error if version is older than year.(Verifying with 11-months limit)
+    def validate_api_version!(version)
+      date = Date.strptime(version, '%Y%m')
+      unless (Date.today - 330) < date
+        msg = LinkedIn::ErrorMessages.unsupported_api_version
+        raise LinkedIn::InvalidRequest.new(msg)
```

Comment:
How about showing a warning in the logs for version older than 11 months so that we can still use the gem and a rise above error for version older than 12 months?

Reply:
Warnings might go unnoticed most of the time, hence handled this into LL-CMS app specs here. https://github.com/marlytics/ll-app/pull/2221/files#diff-46505b050b4fc8f7f1d163937666fd1d5b190427b0270cb6600c50898f441d25R176-R179

Specs will fail in advance to notify developers that its a time to update LinkedIn API version, and will give enough buffer time to update LinkedIn API version, and get it verified properly.